### PR TITLE
[FIX] Tailor, Barn, Bank, Well, Blacksmith Highlights

### PR DESCRIPTION
**Pull request by tiffanydys**
# Description
- Fixes tailor, barn, bank, well, blacksmith highlights on hover

Fixes #issue

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
`yarn test`
Screen Shot 2022-03-23 at 5 39 25 AM

`yarn dev`
## Tailor
Screen Shot 2022-03-22 at 1 11 33 PM

## Barn
Screen Shot 2022-03-22 at 1 14 17 PM

## Well
Screen Shot 2022-03-23 at 1 40 18 AM

## Blacksmith
Screen Shot 2022-03-23 at 1 44 33 AM

## Bank
Screen Shot 2022-03-23 at 1 40 07 AM

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [x] Screenshot if it includes any UI changes
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by jcads** [comment]
I think we also need the Barn House asset so we can highlight the whole building instead of just the button.

cc: @adamhannigan

**Comment by Rofrtd** [comment]
> I think we also need the Barn House asset so we can highlight the whole building instead of just the button.
> 
> cc: @adamhannigan

Yeah, agreed. But I think this could be done on a different PR once we have the asset.

**Comment by Rofrtd** [comment]
Nice fix, can you have a look to do the same highlight on the Craft and Wishing Well labels as well as part of this PR? 

See how they only highlight the asset but not the label..

Screen Shot 2022-03-22 at 11 31 20 pm

Screen Shot 2022-03-22 at 11 32 52 pm

**Comment by tiffanydys** [comment]
> 
> 
> Nice fix, can you have a look to do the same highlight on the Craft and Wishing Well labels as well as part of this PR? 
> 
> 
> 
> See how they only highlight the asset but not the label..
> 
> 
> 
> Screen Shot 2022-03-22 at 11 31 20 pm
> 
> 
> 
> Screen Shot 2022-03-22 at 11 32 52 pm

Hey nice catch! Sure thing. Will edit in a bit, as I am away from my computer. 😆

File: src/features/bank/Bank.tsx [hunk 1]
```diff
@@ -35,23 +35,28 @@ export const Bank: React.FC = () => {
...
-        <Action
-          className="absolute -bottom-6 left-2"
-          text="Bank"
-          icon={token}
-          onClick={open}
+      >
+        <img
+          src={bank}
+          alt="bank"
+          onClick={isNotReadOnly ? openBank : undefined}
```
Comment (thisUsernameAlreadyExists) [inline src/features/bank/Bank.tsx:47]: You can move this check inside `openBank` function and if `isNotReadOnly` you just return before executing anything.

File: src/features/blacksmith/Blacksmith.tsx [hunk 2]
```diff
@@ -35,23 +35,28 @@ export const Blacksmith: React.FC = () => {
...
-        <Action
-          className="absolute -bottom-8 left-1"
-          text="Craft"
-          icon={hammer}
-          onClick={open}
+      >
+        <img
+          src={blacksmith}
+          alt="market"
+          onClick={isNotReadOnly ? openBlacksmith : undefined}
```
Comment (thisUsernameAlreadyExists) [inline src/features/blacksmith/Blacksmith.tsx:47]: Similar to `openBank` situation

File: src/features/wishingWell/WishingWell.tsx [hunk 3]
```diff
@@ -34,23 +34,28 @@ export const WishingWell: React.FC = () => {
...
-        <Action
-          className="absolute -bottom-6 -left-3"
-          text="Wish"
-          icon={icon}
-          onClick={() => open()}
+      >
+        <img
+          src={wishingWell}
+          alt="market"
+          onClick={isNotReadOnly ? openWell : undefined}
```
Comment (thisUsernameAlreadyExists) [inline src/features/wishingWell/WishingWell.tsx:46]: Similar to `openBank` situation

File: src/features/animals/components/Barn.tsx [hunk 4]
```diff
@@ -30,14 +31,21 @@ export const Barn: React.FC = () => {
...
+          "cursor-pointer": isNotReadOnly,
+          "hover:img-highlight": isNotReadOnly,
+        })}
+      >
+        {isNotReadOnly && (
+          <Action
+            className="absolute top-5 left-4"
+            text="Barn"
+            icon={chicken}
+            onClick={() => openBarn()}
```
[the remark this answers is not in the document]
Comment (tiffanydys) [reply src/features/animals/components/Barn.tsx]: Removed arrow function and called directly. Thanks! 😄

File: src/features/bank/Bank.tsx (hunk 1 above)
Comment (tiffanydys) [reply src/features/bank/Bank.tsx]: Moved to openBank function. Thanks! 😄

File: src/features/blacksmith/Blacksmith.tsx (hunk 2 above)
Comment (tiffanydys) [reply src/features/blacksmith/Blacksmith.tsx]: Moved to openBlacksmith function. Thanks! 😄

File: src/features/wishingWell/WishingWell.tsx (hunk 3 above)
Comment (tiffanydys) [reply src/features/wishingWell/WishingWell.tsx]: Moved to openWell function. Thanks! 😄

**Review by spencerdezartsmith — APPROVED** [review]
Great job aligning all of these. 🚀

**Comment by spencerdezartsmith** [comment]
> I think we also need the Barn House asset so we can highlight the whole building instead of just the button.
> 
> cc: @adamhannigan

The barn is currently part of the map. We will need to change this before we can add a highlight.

**Comment by thisUsernameAlreadyExists** [comment]
Great job. Looks good now

**Pull request merged**